db/db_impl_open.cc

```diff
@@ -422,7 +422,8 @@ Status DBImpl::Recover(
     std::vector<std::string> filenames;
     s = env_->GetChildren(immutable_db_options_.wal_dir, &filenames);
     if (!s.ok()) {
-      return s;
+      return Status::InvalidArgument("wal_dir not found",
```

Comment:
`NotFound` or `InvalidArgument`?

Reply:
Ok I see. Actually the return value of `GetChildren` can be `NotFound` or `IOError`. It's possible that we tell the user `wal_dir` is not found but the actual error is permission?

Reply:
Hmm, good catch. I should fix it.